In [1]:
from google.colab import files
uploaded = files.upload()

import pandas as pd

file_name = list(uploaded.keys())[0]
df = pd.read_excel(file_name)

# Reset indeks dan tambahkan 1 untuk membuat indeks dimulai dari 1
df = df.reset_index(drop=True)
df.index = df.index + 1

# Tampilkan beberapa baris pertama DataFrame untuk memastikan data berhasil dimuat
display(df.head())

Saving contoh_data_bermasalah.xlsx to contoh_data_bermasalah.xlsx


ID          Nama  Usia        Kota  Pendapatan       Status Tanggal Daftar
1  118  Pengguna_118    51  Yogyakarta     9819256        Aktif     2023-06-26
2   29   Pengguna_29    37     Jakarta     4991886        Aktif     2023-07-20
3  119  Pengguna_119    31       Medan    19094979        Aktif     2023-05-17
4   44   Pengguna_44    54    Makassar    15843052  Tidak Aktif     2023-05-10
5  124  Pengguna_124    26     Bandung    86377074      Pending     2023-05-05

In [2]:
# Mengecek ukuran dataset (jumlah baris dan kolom)
print("Ukuran dataset (baris, kolom):", df.shape)

# Mengecek jumlah total elemen dalam dataset
print("Jumlah total elemen:", df.size)

Ukuran dataset (baris, kolom): (160, 7)
Jumlah total elemen: 1120


In [3]:
# Mengecek tipe data dari sebuah variabel

my_variable = "Hello"
print("Tipe data my_variable:", type(my_variable))
my_number = 123
print("Tipe data my_number:", type(my_number))

# Mengecek tipe data dari setiap kolom dalam DataFrame

print("\nTipe data setiap kolom dalam DataFrame:")
print(df.dtypes)

Tipe data my_variable: <class 'str'>
Tipe data my_number: <class 'int'>

Tipe data setiap kolom dalam DataFrame:
ID                         int64
Nama                      object
Usia                       int64
Kota                      object
Pendapatan                 int64
Status                    object
Tanggal Daftar    datetime64[ns]
dtype: object


In [4]:
# Mengecek jumlah missing values di setiap kolom
missing_values = df.isnull().sum()

print("Jumlah nilai yang hilang di setiap kolom:")
print(missing_values)

Jumlah nilai yang hilang di setiap kolom:
ID                 0
Nama               0
Usia               0
Kota               0
Pendapatan         0
Status            16
Tanggal Daftar     0
dtype: int64


In [5]:
# Hitung jumlah baris duplikat
duplicate_rows_count = df.duplicated().sum()

# Cetak jumlah baris duplikat
print(f"Jumlah baris duplikat: {duplicate_rows_count}")

# Jika ada baris duplikat, tampilkan beberapa baris duplikat pertama
if duplicate_rows_count > 0:
    print("\nBeberapa baris duplikat pertama:")
    display(df[df.duplicated(keep=False)].head())

Jumlah baris duplikat: 10

Beberapa baris duplikat pertama:


ID          Nama  Usia      Kota  Pendapatan       Status Tanggal Daftar
9    49   Pengguna_49    51  Semarang    18099831        Aktif     2023-11-25
34   63   Pengguna_63    20  Surabaya    12742482        Aktif     2023-05-02
41  146  Pengguna_146    45  Semarang    13493455        Aktif     2023-01-09
42    2    Pengguna_2    45   Bandung     9640118  Tidak Aktif     2023-02-10
54   89   Pengguna_89    27   Bandung    17464535        Aktif     2023-03-29

Dataset yang digunakan memiliki 160 baris dan 7 kolom dengan total 1.120 elemen. Secara umum, tipe data pada setiap kolom sudah sesuai, di mana kolom ID, Usia, dan Pendapatan bertipe integer, kolom Nama, Kota, dan Status bertipe objek atau string, serta kolom Tanggal Daftar bertipe datetime. Namun, hasil pengecekan menunjukkan adanya permasalahan kualitas data, yaitu terdapat 16 nilai yang hilang pada kolom Status dan ditemukan 10 baris data yang bersifat duplikat. Kondisi ini perlu ditangani melalui proses data cleaning, misalnya dengan melakukan imputasi atau pengisian nilai kosong pada kolom Status, serta menghapus baris duplikat agar dataset menjadi lebih bersih dan siap digunakan untuk analisis lebih lanjut.